### DISCUSSION TITLE
System.Data.SqlClient.SqlException (0x80131904): The incoming request has too many parameters. The server supports a maximum of 2100 parameters. Reduce the number of parameters and resend the request.

### DISCUSSION THREAD

**Discussion by robertovaldesperez** [nhibernate/nhibernate-core#2986] [Q&A] [answered]
When making a query that exceeds 2100 parameters, it gives me an error

    select
        (count(*)) as col_0_0_ 
    from
        dbo.promocion developmen0_ 
    where
        (
            developmen0_.referencia in (
                @p0 , @p1 , @p2 , @p3 , @p4 , @p5 , @p6 , @p7 , @p8 , @p9 , @p10 , @p11 , @p12 , @p13 , @p14 , @p15 , @p16 , @p17 , @p18 , @p19 , @p20 , @p21 , @p22 , @p23 , @p24 , @p25 , @p26 , @p27 , @p28 , @p29 , @p30 , @p31 , @p32 , @p33 , @p34 , @p35 , @p36 , @p37 , @p38 , @p39 , @p40 , @p41 , @p42 , @p43 , @p44 , @p45 , @p46 , @p47 , @p48 , @p49 , @p50 , @p51 , @p52 , @p53 , @p54 , @p55 , @p56 , @p57 , @p58 , @p59 , @p60 , @p61 , @p62 , @p63 , @p64 , @p65 , @p66 , @p67 , @p68 , @p69 , @p70 , @p71 , @p72 , @p73 , @p74 , @p75 , @p76 , @p77 , @p78 , @p79 , @p80 , @p81 , @p82 , @p83 , @p84 , @p85 , @p86 , @p87 , @p88 , @p89 , @p90 , @p91 , @p92 , @p93 , @p94 , @p95 , @p96 , @p97 , @p98 , @p99 , @p100 , @p101 , @p102 , @p103 , @p104 , @p105 , @p106 , @p107 , @p108 , @p109 , @p110 , @p111 , @p112 , @p113 , @p114 , @p115 , @p116 , @p117 , @p118 , @p119 , @p120 , @p121 , @p122 , @p123 , @p124 , @p125 , @p126 , @p127 , @p128 , @p129 , @p130 , @p131 , @p132 , @p133 , @p134 , @p135 , @p136 , @p137 , @p138 , @p139 , @p140 , @p141 , @p142 , @p143 , @p144 , @p145 , @p146 , @p147 , @p148 , @p149 , @p150 , @p151 , @p152 , @p153 , @p154 , @p155 , @p156 , @p157 , @p158 , @p159 , @p160 , @p161 , @p162 , @p163 , @p164 , @p165 , @p166 , @p167 , @p168 , @p169 , @p170 , @p171 , @p172 , @p173 , @p174 , @p175 , @p176 , @p177 , @p178 , @p179 , @p180 , @p181 , @p182 , @p183 , @p184 , @p185 , @p186 , @p187 , @p188 , @p189 , @p190 , @p191 , @p192 , @p193 , @p194 , @p195 , @p196 , @p197 , @p198 , @p199 , @p200 , @p201 , @p202 , @p203 , @p204 , @p205 , @p206 , @p207 , @p208 , @p209 , @p210 , @p211 , @p212 , @p213 , @p214 , @p215 , @p216 , @p217 , @p218 , @p219 , @p220 , @p221 , @p222 , @p223 , @p224 , @p225 , @p226 , @p227 , @p228 , @p229 , @p230 , @p231 , @p232 , @p233 , @p234 , @p235 , @p236 , @p237 , @p238 , @p239 , @p240 , @p241 , @p242 , @p243 , @p244 , @p245 , @p246 , @p247 , @p248 , @p249 , @p250 , @p251 , @p252 , @p253 , @p254 , @p255 , @p256 , @p257 , @p258 , @p259 , @p260 , @p261 , @p262 , @p263 , @p264 , @p265 , @p266 , @p267 , @p268 , @p269 , @p270 , @p271 , @p272 , @p273 , @p274 , @p275 , @p276 , @p277 , @p278 , @p279 , @p280 , @p281 , @p282 , @p283 , @p284 , @p285 , @p286 , @p287 , @p288 , @p289 , @p290 , @p291 , @p292 , @p293 , @p294 , @p295 , @p296 , @p297 , @p298 , @p299 , @p300 , @p301 , @p302 , @p303 , @p304 , @p305 , @p306 , @p307 , @p308 , @p309 , @p310 , @p311 , @p312 , @p313 , @p314 , @p315 , @p316 , @p317 , @p318 , @p319 , @p320 , @p321 , @p322 , @p323 , @p324 , @p325 , @p326 , @p327 , @p328 , @p329 , @p330 , @p331 , @p332 , @p333 , @p334 , @p335 , @p336 , @p337 , @p338 , @p339 , @p340 , @p341 , @p342 , @p343 , @p344 , @p345 , @p346 , @p347 , @p348 , @p349 , @p350 , @p351 , @p352 , @p353 , @p354 , @p355 , @p356 , @p357 , @p358 , @p359 , @p360 , @p361 , @p362 , @p363 , @p364 , @p365 , @p366 , @p367 , @p368 , @p369 , @p370 , @p371 , @p372 , @p373 , @p374 , @p375 , @p376 , @p377 , @p378 , @p379 , @p380 , @p381 , @p382 , @p383 , @p384 , @p385 , @p386 , @p387 , @p388 , @p389 , @p390 , @p391 , @p392 , @p393 , @p394 , @p395 , @p396 , @p397 , @p398 , @p399 , @p400 , @p401 , @p402 , @p403 , @p404 , @p405 , @p406 , @p407 , @p408 , @p409 , @p410 , @p411 , @p412 , @p413 , @p414 , @p415 , @p416 , @p417 , @p418 , @p419 , @p420 , @p421 , @p422 , @p423 , @p424 , @p425 , @p426 , @p427 , @p428 , @p429 , @p430 , @p431 , @p432 , @p433 , @p434 , @p435 , @p436 , @p437 , @p438 , @p439 , @p440 , @p441 , @p442 , @p443 , @p444 , @p445 , @p446 , @p447 , @p448 , @p449 , @p450 , @p451 , @p452 , @p453 , @p454 , @p455 , @p456 , @p457 , @p458 , @p459 , @p460 , @p461 , @p462 , @p463 , @p464 , @p465 , @p466 , @p467 , @p468 , @p469 , @p470 , @p471 , @p472 , @p473 , @p474 , @p475 , @p476 , @p477 , @p478 , @p479 , @p480 , @p481 , @p482 , @p483 , @p484 , @p485 , @p486 , @p487 , @p488 , @p489 , @p490 , @p491 , @p492 , @p493 , @p494 , @p495 , @p496 , @p497 , @p498 , @p499 , @p500 , @p501 , @p502 , @p503 , @p504 , @p505 , @p506 , @p507 , @p508 , @p509 , @p510 , @p511 , @p512 , @p513 , @p514 , @p515 , @p516 , @p517 , @p518 , @p519 , @p520 , @p521 , @p522 , @p523 , @p524 , @p525 , @p526 , @p527 , @p528 , @p529 , @p530 , @p531 , @p532 , @p533 , @p534 , @p535 , @p536 , @p537 , @p538 , @p539 , @p540 , @p541 , @p542 , @p543 , @p544 , @p545 , @p546 , @p547 , @p548 , @p549 , @p550 , @p551 , @p552 , @p553 , @p554 , @p555 , @p556 , @p557 , @p558 , @p559 , @p560 , @p561 , @p562 , @p563 , @p564 , @p565 , @p566 , @p567 , @p568 , @p569 , @p570 , @p571 , @p572 , @p573 , @p574 , @p575 , @p576 , @p577 , @p578 , @p579 , @p580 , @p581 , @p582 , @p583 , @p584 , @p585 , @p586 , @p587 , @p588 , @p589 , @p590 , @p591 , @p592 , @p593 , @p594 , @p595 , @p596 , @p597 , @p598 , @p599 , @p600 , @p601 , @p602 , @p603 , @p604 , @p605 , @p606 , @p607 , @p608 , @p609 , @p610 , @p611 , @p612 , @p613 , @p614 , @p615 , @p616 , @p617 , @p618 , @p619 , @p620 , @p621 , @p622 , @p623 , @p624 , @p625 , @p626 , @p627 , @p628 , @p629 , @p630 , @p631 , @p632 , @p633 , @p634 , @p635 , @p636 , @p637 , @p638 , @p639 , @p640 , @p641 , @p642 , @p643 , @p644 , @p645 , @p646 , @p647 , @p648 , @p649 , @p650 , @p651 , @p652 , @p653 , @p654 , @p655 , @p656 , @p657 , @p658 , @p659 , @p660 , @p661 , @p662 , @p663 , @p664 , @p665 , @p666 , @p667 , @p668 , @p669 , @p670 , @p671 , @p672 , @p673 , @p674 , @p675 , @p676 , @p677 , @p678 , @p679 , @p680 , @p681 , @p682 , @p683 , @p684 , @p685 , @p686 , @p687 , @p688 , @p689 , @p690 , @p691 , @p692 , @p693 , @p694 , @p695 , @p696 , @p697 , @p698 , @p699 , @p700 , @p701 , @p702 , @p703 , @p704 , @p705 , @p706 , @p707 , @p708 , @p709 , @p710 , @p711 , @p712 , @p713 , @p714 , @p715 , @p716 , @p717 , @p718 , @p719 , @p720 , @p721 , @p722 , @p723 , @p724 , @p725 , @p726 , @p727 , @p728 , @p729 , @p730 , @p731 , @p732 , @p733 , @p734 , @p735 , @p736 , @p737 , @p738 , @p739 , @p740 , @p741 , @p742 , @p743 , @p744 , @p745 , @p746 , @p747 , @p748 , @p749 , @p750 , @p751 , @p752 , @p753 , @p754 , @p755 , @p756 , @p757 , @p758 , @p759 , @p760 , @p761 , @p762 , @p763 , @p764 , @p765 , @p766 , @p767 , @p768 , @p769 , @p770 , @p771 , @p772 , @p773 , @p774 , @p775 , @p776 , @p777 , @p778 , @p779 , @p780 , @p781 , @p782 , @p783 , @p784 , @p785 , @p786 , @p787 , @p788 , @p789 , @p790 , @p791 , @p792 , @p793 , @p794 , @p795 , @p796 , @p797 , @p798 , @p799 , @p800 , @p801 , @p802 , @p803 , @p804 , @p805 , @p806 , @p807 , @p808 , @p809 , @p810 , @p811 , @p812 , @p813 , @p814 , @p815 , @p816 , @p817 , @p818 , @p819 , @p820 , @p821 , @p822 , @p823 , @p824 , @p825 , @p826 , @p827 , @p828 , @p829 , @p830 , @p831 , @p832 , @p833 , @p834 , @p835 , @p836 , @p837 , @p838 , @p839 , @p840 , @p841 , @p842 , @p843 , @p844 , @p845 , @p846 , @p847 , @p848 , @p849 , @p850 , @p851 , @p852 , @p853 , @p854 , @p855 , @p856 , @p857 , @p858 , @p859 , @p860 , @p861 , @p862 , @p863 , @p864 , @p865 , @p866 , @p867 , @p868 , @p869 , @p870 , @p871 , @p872 , @p873 , @p874 , @p875 , @p876 , @p877 , @p878 , @p879 , @p880 , @p881 , @p882 , @p883 , @p884 , @p885 , @p886 , @p887 , @p888 , @p889 , @p890 , @p891 , @p892 , @p893 , @p894 , @p895 , @p896 , @p897 , @p898 , @p899 , @p900 , @p901 , @p902 , @p903 , @p904 , @p905 , @p906 , @p907 , @p908 , @p909 , @p910 , @p911 , @p912 , @p913 , @p914 , @p915 , @p916 , @p917 , @p918 , @p919 , @p920 , @p921 , @p922 , @p923 , @p924 , @p925 , @p926 , @p927 , @p928 , @p929 , @p930 , @p931 , @p932 , @p933 , @p934 , @p935 , @p936 , @p937 , @p938 , @p939 , @p940 , @p941 , @p942 , @p943 , @p944 , @p945 , @p946 , @p947 , @p948 , @p949 , @p950 , @p951 , @p952 , @p953 , @p954 , @p955 , @p956 , @p957 , @p958 , @p959 , @p960 , @p961 , @p962 , @p963 , @p964 , @p965 , @p966 , @p967 , @p968 , @p969 , @p970 , @p971 , @p972 , @p973 , @p974 , @p975 , @p976 , @p977 , @p978 , @p979 , @p980 , @p981 , @p982 , @p983 , @p984 , @p985 , @p986 , @p987 , @p988 , @p989 , @p990 , @p991 , @p992 , @p993 , @p994 , @p995 , @p996 , @p997 , @p998 , @p999 , @p1000 , @p1001 , @p1002 , @p1003 , @p1004 , @p1005 , @p1006 , @p1007 , @p1008 , @p1009 , @p1010 , @p1011 , @p1012 , @p1013 , @p1014 , @p1015 , @p1016 , @p1017 , @p1018 , @p1019 , @p1020 , @p1021 , @p1022 , @p1023 , @p1024 , @p1025 , @p1026 , @p1027 , @p1028 , @p1029 , @p1030 , @p1031 , @p1032 , @p1033 , @p1034 , @p1035 , @p1036 , @p1037 , @p1038 , @p1039 , @p1040 , @p1041 , @p1042 , @p1043 , @p1044 , @p1045 , @p1046 , @p1047 , @p1048 , @p1049 , @p1050 , @p1051 , @p1052 , @p1053 , @p1054 , @p1055 , @p1056 , @p1057 , @p1058 , @p1059 , @p1060 , @p1061 , @p1062 , @p1063 , @p1064 , @p1065 , @p1066 , @p1067 , @p1068 , @p1069 , @p1070 , @p1071 , @p1072 , @p1073 , @p1074 , @p1075 , @p1076 , @p1077 , @p1078 , @p1079 , @p1080 , @p1081 , @p1082 , @p1083 , @p1084 , @p1085 , @p1086 , @p1087 , @p1088 , @p1089 , @p1090 , @p1091 , @p1092 , @p1093 , @p1094 , @p1095 , @p1096 , @p1097 , @p1098 , @p1099 , @p1100 , @p1101 , @p1102 , @p1103 , @p1104 , @p1105 , @p1106 , @p1107 , @p1108 , @p1109 , @p1110 , @p1111 , @p1112 , @p1113 , @p1114 , @p1115 , @p1116 , @p1117 , @p1118 , @p1119 , @p1120 , @p1121 , @p1122 , @p1123 , @p1124 , @p1125 , @p1126 , @p1127 , @p1128 , @p1129 , @p1130 , @p1131 , @p1132 , @p1133 , @p1134 , @p1135 , @p1136 , @p1137 , @p1138 , @p1139 , @p1140 , @p1141 , @p1142 , @p1143 , @p1144 , @p1145 , @p1146 , @p1147 , @p1148 , @p1149 , @p1150 , @p1151 , @p1152 , @p1153 , @p1154 , @p1155 , @p1156 , @p1157 , @p1158 , @p1159 , @p1160 , @p1161 , @p1162 , @p1163 , @p1164 , @p1165 , @p1166 , @p1167 , @p1168 , @p1169 , @p1170 , @p1171 , @p1172 , @p1173 , @p1174 , @p1175 , @p1176 , @p1177 , @p1178 , @p1179 , @p1180 , @p1181 , @p1182 , @p1183 , @p1184 , @p1185 , @p1186 , @p1187 , @p1188 , @p1189 , @p1190 , @p1191 , @p1192 , @p1193 , @p1194 , @p1195 , @p1196 , @p1197 , @p1198 , @p1199 , @p1200 , @p1201 , @p1202 , @p1203 , @p1204 , @p1205 , @p1206 , @p1207 , @p1208 , @p1209 , @p1210 , @p1211 , @p1212 , @p1213 , @p1214 , @p1215 , @p1216 , @p1217 , @p1218 , @p1219 , @p1220 , @p1221 , @p1222 , @p1223 , @p1224 , @p1225 , @p1226 , @p1227 , @p1228 , @p1229 , @p1230 , @p1231 , @p1232 , @p1233 , @p1234 , @p1235 , @p1236 , @p1237 , @p1238 , @p1239 , @p1240 , @p1241 , @p1242 , @p1243 , @p1244 , @p1245 , @p1246 , @p1247 , @p1248 , @p1249 , @p1250 , @p1251 , @p1252 , @p1253 , @p1254 , @p1255 , @p1256 , @p1257 , @p1258 , @p1259 , @p1260 , @p1261 , @p1262 , @p1263 , @p1264 , @p1265 , @p1266 , @p1267 , @p1268 , @p1269 , @p1270 , @p1271 , @p1272 , @p1273 , @p1274 , @p1275 , @p1276 , @p1277 , @p1278 , @p1279 , @p1280 , @p1281 , @p1282 , @p1283 , @p1284 , @p1285 , @p1286 , @p1287 , @p1288 , @p1289 , @p1290 , @p1291 , @p1292 , @p1293 , @p1294 , @p1295 , @p1296 , @p1297 , @p1298 , @p1299 , @p1300 , @p1301 , @p1302 , @p1303 , @p1304 , @p1305 , @p1306 , @p1307 , @p1308 , @p1309 , @p1310 , @p1311 , @p1312 , @p1313 , @p1314 , @p1315 , @p1316 , @p1317 , @p1318 , @p1319 , @p1320 , @p1321 , @p1322 , @p1323 , @p1324 , @p1325 , @p1326 , @p1327 , @p1328 , @p1329 , @p1330 , @p1331 , @p1332 , @p1333 , @p1334 , @p1335 , @p1336 , @p1337 , @p1338 , @p1339 , @p1340 , @p1341 , @p1342 , @p1343 , @p1344 , @p1345 , @p1346 , @p1347 , @p1348 , @p1349 , @p1350 , @p1351 , @p1352 , @p1353 , @p1354 , @p1355 , @p1356 , @p1357 , @p1358 , @p1359 , @p1360 , @p1361 , @p1362 , @p1363 , @p1364 , @p1365 , @p1366 , @p1367 , @p1368 , @p1369 , @p1370 , @p1371 , @p1372 , @p1373 , @p1374 , @p1375 , @p1376 , @p1377 , @p1378 , @p1379 , @p1380 , @p1381 , @p1382 , @p1383 , @p1384 , @p1385 , @p1386 , @p1387 , @p1388 , @p1389 , @p1390 , @p1391 , @p1392 , @p1393 , @p1394 , @p1395 , @p1396 , @p1397 , @p1398 , @p1399 , @p1400 , @p1401 , @p1402 , @p1403 , @p1404 , @p1405 , @p1406 , @p1407 , @p1408 , @p1409 , @p1410 , @p1411 , @p1412 , @p1413 , @p1414 , @p1415 , @p1416 , @p1417 , @p1418 , @p1419 , @p1420 , @p1421 , @p1422 , @p1423 , @p1424 , @p1425 , @p1426 , @p1427 , @p1428 , @p1429 , @p1430 , @p1431 , @p1432 , @p1433 , @p1434 , @p1435 , @p1436 , @p1437 , @p1438 , @p1439 , @p1440 , @p1441 , @p1442 , @p1443 , @p1444 , @p1445 , @p1446 , @p1447 , @p1448 , @p1449 , @p1450 , @p1451 , @p1452 , @p1453 , @p1454 , @p1455 , @p1456 , @p1457 , @p1458 , @p1459 , @p1460 , @p1461 , @p1462 , @p1463 , @p1464 , @p1465 , @p1466 , @p1467 , @p1468 , @p1469 , @p1470 , @p1471 , @p1472 , @p1473 , @p1474 , @p1475 , @p1476 , @p1477 , @p1478 , @p1479 , @p1480 , @p1481 , @p1482 , @p1483 , @p1484 , @p1485 , @p1486 , @p1487 , @p1488 , @p1489 , @p1490 , @p1491 , @p1492 , @p1493 , @p1494 , @p1495 , @p1496 , @p1497 , @p1498 , @p1499 , @p1500 , @p1501 , @p1502 , @p1503 , @p1504 , @p1505 , @p1506 , @p1507 , @p1508 , @p1509 , @p1510 , @p1511 , @p1512 , @p1513 , @p1514 , @p1515 , @p1516 , @p1517 , @p1518 , @p1519 , @p1520 , @p1521 , @p1522 , @p1523 , @p1524 , @p1525 , @p1526 , @p1527 , @p1528 , @p1529 , @p1530 , @p1531 , @p1532 , @p1533 , @p1534 , @p1535 , @p1536 , @p1537 , @p1538 , @p1539 , @p1540 , @p1541 , @p1542 , @p1543 , @p1544 , @p1545 , @p1546 , @p1547 , @p1548 , @p1549 , @p1550 , @p1551 , @p1552 , @p1553 , @p1554 , @p1555 , @p1556 , @p1557 , @p1558 , @p1559 , @p1560 , @p1561 , @p1562 , @p1563 , @p1564 , @p1565 , @p1566 , @p1567 , @p1568 , @p1569 , @p1570 , @p1571 , @p1572 , @p1573 , @p1574 , @p1575 , @p1576 , @p1577 , @p1578 , @p1579 , @p1580 , @p1581 , @p1582 , @p1583 , @p1584 , @p1585 , @p1586 , @p1587 , @p1588 , @p1589 , @p1590 , @p1591 , @p1592 , @p1593 , @p1594 , @p1595 , @p1596 , @p1597 , @p1598 , @p1599 , @p1600 , @p1601 , @p1602 , @p1603 , @p1604 , @p1605 , @p1606 , @p1607 , @p1608 , @p1609 , @p1610 , @p1611 , @p1612 , @p1613 , @p1614 , @p1615 , @p1616 , @p1617 , @p1618 , @p1619 , @p1620 , @p1621 , @p1622 , @p1623 , @p1624 , @p1625 , @p1626 , @p1627 , @p1628 , @p1629 , @p1630 , @p1631 , @p1632 , @p1633 , @p1634 , @p1635 , @p1636 , @p1637 , @p1638 , @p1639 , @p1640 , @p1641 , @p1642 , @p1643 , @p1644 , @p1645 , @p1646 , @p1647 , @p1648 , @p1649 , @p1650 , @p1651 , @p1652 , @p1653 , @p1654 , @p1655 , @p1656 , @p1657 , @p1658 , @p1659 , @p1660 , @p1661 , @p1662 , @p1663 , @p1664 , @p1665 , @p1666 , @p1667 , @p1668 , @p1669 , @p1670 , @p1671 , @p1672 , @p1673 , @p1674 , @p1675 , @p1676 , @p1677 , @p1678 , @p1679 , @p1680 , @p1681 , @p1682 , @p1683 , @p1684 , @p1685 , @p1686 , @p1687 , @p1688 , @p1689 , @p1690 , @p1691 , @p1692 , @p1693 , @p1694 , @p1695 , @p1696 , @p1697 , @p1698 , @p1699 , @p1700 , @p1701 , @p1702 , @p1703 , @p1704 , @p1705 , @p1706 , @p1707 , @p1708 , @p1709 , @p1710 , @p1711 , @p1712 , @p1713 , @p1714 , @p1715 , @p1716 , @p1717 , @p1718 , @p1719 , @p1720 , @p1721 , @p1722 , @p1723 , @p1724 , @p1725 , @p1726 , @p1727 , @p1728 , @p1729 , @p1730 , @p1731 , @p1732 , @p1733 , @p1734 , @p1735 , @p1736 , @p1737 , @p1738 , @p1739 , @p1740 , @p1741 , @p1742 , @p1743 , @p1744 , @p1745 , @p1746 , @p1747 , @p1748 , @p1749 , @p1750 , @p1751 , @p1752 , @p1753 , @p1754 , @p1755 , @p1756 , @p1757 , @p1758 , @p1759 , @p1760 , @p1761 , @p1762 , @p1763 , @p1764 , @p1765 , @p1766 , @p1767 , @p1768 , @p1769 , @p1770 , @p1771 , @p1772 , @p1773 , @p1774 , @p1775 , @p1776 , @p1777 , @p1778 , @p1779 , @p1780 , @p1781 , @p1782 , @p1783 , @p1784 , @p1785 , @p1786 , @p1787 , @p1788 , @p1789 , @p1790 , @p1791 , @p1792 , @p1793 , @p1794 , @p1795 , @p1796 , @p1797 , @p1798 , @p1799 , @p1800 , @p1801 , @p1802 , @p1803 , @p1804 , @p1805 , @p1806 , @p1807 , @p1808 , @p1809 , @p1810 , @p1811 , @p1812 , @p1813 , @p1814 , @p1815 , @p1816 , @p1817 , @p1818 , @p1819 , @p1820 , @p1821 , @p1822 , @p1823 , @p1824 , @p1825 , @p1826 , @p1827 , @p1828 , @p1829 , @p1830 , @p1831 , @p1832 , @p1833 , @p1834 , @p1835 , @p1836 , @p1837 , @p1838 , @p1839 , @p1840 , @p1841 , @p1842 , @p1843 , @p1844 , @p1845 , @p1846 , @p1847 , @p1848 , @p1849 , @p1850 , @p1851 , @p1852 , @p1853 , @p1854 , @p1855 , @p1856 , @p1857 , @p1858 , @p1859 , @p1860 , @p1861 , @p1862 , @p1863 , @p1864 , @p1865 , @p1866 , @p1867 , @p1868 , @p1869 , @p1870 , @p1871 , @p1872 , @p1873 , @p1874 , @p1875 , @p1876 , @p1877 , @p1878 , @p1879 , @p1880 , @p1881 , @p1882 , @p1883 , @p1884 ....

**Comment by oskarb** [comment, marked as answer]
Yes, and the exception is telling you that this is a limitation in
SQL Server and what to do about it (i.e. reduce the number of parameters).

**Reply by guangzhengli** [reply]
Can you confirm whether there are plans to support automatic batch operations for SQL statements with more than 2100 parameters? For example, if a query has 2200 parameters, it would automatically be split into two separate query.

Or Is there currently a solution available for this, or does it still rely on developers to resolve this issue?

**Reply by fredericDelaporte** [reply]
There are no such plans. It could seem simple to do, but that is not the case. By example, how to handle paging if a query appears to be paged? How the ordering of rows between the split queries would be handled ? The ORM has no way to reliably know if splitting the query would yield the intended results, or something else. Only the developer can do this.
